dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -423,3 +459,224 @@ def __init__(self,
                                                          squeeze_labels=squeeze_labels,
                                                          dynamic_shape=dynamic_shape,
                                                          last_batch_padded = last_batch_padded)
+
+###############################################
+###############################################
+################## Gluon API ##################
+###############################################
+###############################################
+
+
+class SmartArray(object):
+    def __init__(self, array):
+        self._data = array.reshape(-1)
+        self._view = array
+
+    def resize(self, shape):
+        new_size = np.prod(shape)
+
+        if new_size > self._data.size:
+            self._data = mx.nd.zeros(new_size, self._data.context, dtype=self._data.dtype)
+            self._view = self._data
+        elif new_size < self._data.size:
+            self._view = self._data[:new_size]
+        else:
+            self._view = self._data
+        self._view = self._view.reshape(shape)
+        return self._view
+
+    @property
+    def array(self):
+        return self._view
+
+
+class DALIGluonIterator(_DALIIteratorBase):
+    """
+    General DALI iterator for MXNet with Gluon API. It can return any number of
+    outputs from the DALI pipeline in the form of per GPU tuples of NDArray
+    of NDArrays or list of NDArrays (for output marked as DALIGluonIterator.SPARSE_TAG).
+
```

Comment:
I'm not sure that I follow. It sounds like we have a tuple of NDArrays (one NDArray per GPU), and that NDArray can have `NDArrays` or `list of NDArrays` as elements. Or it is one tuple per GPU which has elements being `NDArray of NDArrays` or `list of NDArrays`.

Reply:
the former

>  of NDArray of NDArrays 

this is a typo :stuck_out_tongue: 

I tried to clarify that in the latest commit